pkg/workloads/containerd/client.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package containerd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/docker/engine-api/client"
+	dTypes "github.com/docker/engine-api/types"
+	dNetwork "github.com/docker/engine-api/types/network"
+)
+
+var (
+	dockerClient *client.Client
+)
+
+// Client returns the default containerd client
+func Client() *client.Client {
+	return dockerClient
+}
+
+// Init initializes the containerd package
+func Init(endpoint string) error {
+	defaultHeaders := map[string]string{"User-Agent": "cilium"}
+	c, err := client.NewClient(endpoint, "v1.21", nil, defaultHeaders)
+	if err != nil {
+		return err
+	}
+
+	dockerClient = c
+
+	return nil
+}
+
+// InitMock initializes the containerd packged with a mocked runtime
+func InitMock() error {
```

Comment:
Does this (L65-L138) needs to be in here? Can't this be in `_test.go`?

Reply:
I don't mind it being in `client.go`. Keeps the related code together.

Reply:
But you could have the `client_test.go`

Reply:
How? The code will no longer be included.

Reply:
I thought InitMock was not being used outside the `containerd` pkg